src/CoreWCF.Primitives/src/CoreWCF/Description/WsdlHelper.cs

```diff
@@ -0,0 +1,328 @@
+            {
+                XmlNode policyId = policyElement.Attributes.GetNamedItem(MetadataStrings.Wsu.Attributes.Id, MetadataStrings.Wsu.NamespaceUri);
+                string policyIdString = policyId.Value;
+                string policyIdStringWithOldBindingName = string.Format(CultureInfo.InvariantCulture, policyIdStringPrefixFormat, oldBindingName);
+                string policyIdStringWithNewBindingName = string.Format(CultureInfo.InvariantCulture, policyIdStringPrefixFormat, newBindingName);
+                if (policyId != null && policyIdString != null && policyIdString.StartsWith(policyIdStringWithOldBindingName, StringComparison.Ordinal))
+                {
+                    policyId.Value = policyIdStringWithNewBindingName + policyIdString.Substring(policyIdStringWithOldBindingName.Length);
+                }
+            }
+
+            // policy reference
+            UpdatePolicyReference(binding.Extensions, oldBindingName, newBindingName);
+            foreach (WsdlNS.OperationBinding operationBinding in binding.Operations)
+            {
+                UpdatePolicyReference(operationBinding.Extensions, oldBindingName, newBindingName);
+                if (operationBinding.Input != null)
+                {
+                    UpdatePolicyReference(operationBinding.Input.Extensions, oldBindingName, newBindingName);
+                }
+
+                if (operationBinding.Output != null)
+                {
+                    UpdatePolicyReference(operationBinding.Output.Extensions, oldBindingName, newBindingName);
+                }
+
+                foreach (WsdlNS.FaultBinding fault in operationBinding.Faults)
+                {
+                    UpdatePolicyReference(fault.Extensions, oldBindingName, newBindingName);
+                }
+            }
+        }
+
+        private static void UpdatePolicyReference(WsdlNS.ServiceDescriptionFormatExtensionCollection extensions, string oldBindingName, string newBindingName)
+        {
+            IEnumerable<XmlElement> bindingPolicyReferences = FindAllElements(extensions, MetadataStrings.WSPolicy.Elements.PolicyReference);
+            string policyReferencePrefixFormat = "#{0}_";
+            foreach (XmlElement policyReferenceElement in bindingPolicyReferences)
+            {
+                XmlNode policyReference = policyReferenceElement.Attributes.GetNamedItem(MetadataStrings.WSPolicy.Attributes.URI);
+                string policyReferenceValue = policyReference.Value;
+                string policyReferenceValueWithOldBindingName = string.Format(CultureInfo.InvariantCulture, policyReferencePrefixFormat, oldBindingName);
+                string policyReferenceValueWithNewBindingName = string.Format(CultureInfo.InvariantCulture, policyReferencePrefixFormat, newBindingName);
+                if (policyReference != null && policyReferenceValue != null && policyReferenceValue.StartsWith(policyReferenceValueWithOldBindingName, StringComparison.Ordinal))
```

Comment:
`policyReference` cannot be null here due to `string policyReferenceValue = policyReference.Value;` on line 196

Reply:
Same pattern as policyId. I'll do the same fix, use null coalescence when setting policyReferenceValue.